src/vignette/api/legacy/routes.clj

```diff
@@ -68,10 +78,11 @@
   (= (:zone map) (str "/" archive-dir)))
 
 (defn zone [map]
-  (if (and (:zone map)
+  (if (and (not (empty? (:zone map)))
            (not (archive? map)))
-    (let [[_ zone] (re-find #"^/([a-z]+)$" (get map :zone ""))]
-      zone)
+    (if (= (subs (:zone map) 0 1) "/")
+      (subs (:zone map) 1)
+      (:zone map))
```

Comment:
A comment about this would be helpful. From the outside it's not clear what is happening.